01_basic_commands/020_search_files/search_files.md

```diff
@@ -1 +1,139 @@
+| -iname *pattern* |Case-insensitive *-name* option                                                         |
+| -inum *n*        |Files with Inode number *n*                                                             |
+| -mmin *n*        |Files/Directories, with content last modified *n* minutes ago                           |
+| -mtime *n*       |Files/Directories, with content last modified *n*\*24 minutes ago                       |
+| -name *pattern*  |Files/Directories with specified wildcard *pattern*                                     |
+| -newer *file*    |Files/Directories with content more recently modified than *file*                       |
+| -nouser          |Files/Directories belonging to no valid user                                            |
+| -nogroup         |Files/Directories belonging to no valid group                                           |
+| -perm *mode*     |Files/Directories with permissions set to *mode* (given as octal or symbolic notation)  |
+| -samefile *name* |Files that share sae inode number as file *name*                                         |
+| -size *n*        |Files of size *n*                                                                       |
+| -type *c*        |Files of type *c* (types listed in table below)                                         |
+| -user *name*     |Files/Directories belonging to user *name* (given as username or user ID)               |  
+
+| File Type | Description                   |
+|---        | ---                           |
+|b          |Block special device file      |
+|c          |Character special device file  |
+|d          |Directory                      |
+|f          |Regular File                   |
+|l          |Symbolic Link                  |
+
+### Logical Operators
+The `find` command allows combination of tests via logical operators to search for files more granular.
+
+| **Operator**  | **Description**                                                                       |
+|---            |---                                                                                    |
+| -and/-a       | True if the tests on both sides are true (Implied as default, when no operator used)  |
+| -or/-o        | True if one test on the side is true                                                  |
+| -not/!        | True if the test following is false                                                   |
+| ( )           | Groups operations, modifying the evaluation order                                     |
+
+The example below searches for *png* and *pdf* (`-name "*.png" -or -name "*.pdf"`; `-type f`) files, in the home dir of the current user (`~`), owned by the use *liam* (`-user dummy`)
+
+``` bash
+find ~ \( -name "*.png" -or -name "*.pdf" \) -type f -user dummy
+```
+
+### Actions
+
+Find can be used to execute actions on the files found. 
+
+#### Predefined Actions
```

Comment:
You don't have to reinvent the wheel. Give hints to the chapter in "The Linux command line", "--help" pages or the "man pages". Try not the repeat content which is already available there. Only give pointers what the student have to read. Focus yourself on simple practical exercises and a sample solution.